fusillade/clouddirectory.py

```diff
@@ -1627,31 +1636,27 @@ def get_users_page(self, next_token=None, per_page=None) -> Tuple[Dict, str]:
 
     def add_users(self, users: List['User']) -> None:
         if users:
-            operations = [
-                self.cd.batch_attach_typed_link(
-                    i.object_ref,
-                    self.object_ref,
-                    'membership_link',
-                    {'member_of': 'group'}
-                )
-                for i in users]
+            operations = []
+            [operations.extend(user.add_groups(groups=[self.name], run=False)) for user in users]
             self.cd.batch_write(operations)
             logger.info(dict(message="Adding users to group",
                              object=dict(type=self.object_type, path_name=self._path_name),
                              users=[user._path_name for user in users]))
 
-    def remove_users(self, users: List[str]) -> None:
+    def remove_users(self, users: List['User']) -> None:
         """
         Removes users from this group.
 
         :param users: a list of user names to remove from group
         :return:
         """
-        for user in users:
-            User(user).remove_groups([self._path_name])
-        logger.info(dict(message="Removing users from group",
-                         object=dict(type=self.object_type, path_name=self._path_name),
-                         users=[user for user in users]))
+        if users:
+            operations = []
+            [operations.extend(user.remove_groups(groups=[self.name], run=False)) for user in users]
```

Comment:
This list comprehension is stored in memory and then discarded. Is there a better implementation?